controllers/gslb_controller.go

```diff
@@ -127,42 +129,49 @@ func (r *GslbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	// Add finalizer for this CR
 	if !contains(gslb.GetFinalizers(), gslbFinalizer) {
 		if err := r.addFinalizer(gslb); err != nil {
+			m.ErrorIncrement(gslb)
 			return result.RequeueError(err)
 		}
 	}
 
 	// == Ingress ==========
 	ingress, err := r.gslbIngress(gslb)
 	if err != nil {
+		m.ErrorIncrement(gslb)
 		return result.RequeueError(err)
 	}
 
 	err = r.saveIngress(gslb, ingress)
 	if err != nil {
+		m.ErrorIncrement(gslb)
 		return result.RequeueError(err)
 	}
 
 	// == external-dns dnsendpoints CRs ==
 	dnsEndpoint, err := r.gslbDNSEndpoint(gslb)
 	if err != nil {
+		m.ErrorIncrement(gslb)
 		return result.RequeueError(err)
 	}
 
 	err = r.DNSProvider.SaveDNSEndpoint(gslb, dnsEndpoint)
 	if err != nil {
+		m.ErrorIncrement(gslb)
 		return result.RequeueError(err)
 	}
 
 	// == handle delegated zone in Edge DNS
 	err = r.DNSProvider.CreateZoneDelegationForExternalDNS(gslb)
 	if err != nil {
 		log.Err(err).Msg("Unable to create zone delegation")
+		m.ErrorIncrement(gslb)
 		return result.Requeue()
 	}
 
 	// == Status =
 	err = r.updateGslbStatus(gslb)
 	if err != nil {
+		m.ErrorIncrement(gslb)
```

Comment:
Looks like an emerging standard pattern of k8gb specific error handling. Should we briefly document it in CONTRIBUTING.md?

Reply:
@ytsarev , thx, amended